### PR TITLE
Feature/ui mapreduce

### PR DESCRIPTION
**Pull request by elfenheart**
ToDo: 
- Testing on Distributed

### CODE REVIEW

File: cdap-ui/app/features/mapreduce/controllers/tabs/status-ctrl.js [hunk 1]
```diff
@@ -1 +1,85 @@
...
+      var aggregate = 0;
+      angular.forEach(tasks, function (task) {
+        if(task.state === 'SUCCEEDED') {
+          aggregate += task.progress;
+        }
+      });
+      return aggregate ? ((aggregate / (tasks.length)) * 100).toFixed(1) : 0;
+    };
+
+    $scope.getFailedTasks = function (tasks) {
```
Comment (anwar6953) [inline cdap-ui/app/features/mapreduce/controllers/tabs/status-ctrl.js]: This seems redundant to the computation below (lines 55-80), where it computes the stats.failed there also.
Or is there a difference?

File: cdap-ui/app/features/mapreduce/controllers/tabs/runs/runs-detail-ctrl.js [hunk 2]
```diff
@@ -8,4 +10,72 @@ angular.module(PKG.name + '.feature.mapreduce')
...
+      $scope.info = res;
+      $scope.mapProgress = Math.floor(res.mapProgress * 100);
+      $scope.reduceProgress = Math.floor(res.reduceProgress * 100);
+
+      $scope.mapperStats = getStats($scope.info.mapTasks);
+      $scope.reducerStats = getStats($scope.info.reduceTasks);
+    });
+
+
+    $scope.getCompletedPercentage = function(tasks) {
```
Comment (anwar6953) [inline cdap-ui/app/features/mapreduce/controllers/tabs/runs/runs-detail-ctrl.js]: where's this used?

File: cdap-ui/app/features/mapreduce/controllers/tabs/runs/runs-detail-ctrl.js [hunk 3]
```diff
@@ -8,4 +10,72 @@ angular.module(PKG.name + '.feature.mapreduce')
...
+        pending: 0,
+        killed: 0,
+        failed: 0,
+        recordsIn: 0,
+        bytesIn: 0,
+        total: 0
+      };
+
+      angular.forEach(tasks, function (task) {
+        if (task.state === 'SUCCEEDED') {
```
Comment (ajainarayanan) [inline cdap-ui/app/features/mapreduce/controllers/tabs/runs/runs-detail-ctrl.js]: Can we use a switch statement here?

File: cdap-ui/app/features/mapreduce/controllers/tabs/status-ctrl.js [hunk 4]
```diff
@@ -1 +1,85 @@
...
+        basePath = '/apps/' +
+            $state.params.appId +
+            '/mapreduce/' +
+            $state.params.programId;
+
+    dataSrc.request({
+      _cdapNsPath: basePath + '/runs'
+    })
+    .then(function(runs) {
+      $scope.latestRun = runs[0];
```
Comment (ajainarayanan) [inline cdap-ui/app/features/mapreduce/controllers/tabs/status-ctrl.js]: What happens when we have no runs?

File: cdap-ui/app/features/mapreduce/controllers/tabs/runs/runs-detail-ctrl.js (hunk 2 above)
Comment (elfenheart) [inline cdap-ui/app/features/mapreduce/controllers/tabs/runs/runs-detail-ctrl.js]: being used to compute percentage while progress is being fixed in the backend

**Comment by ajainarayanan** [comment]
:+1: LGTM

**Pull request merged**